tests/test_riscv.c

```diff
@@ -0,0 +1,159 @@
+
+#include <capstone/platform.h>
+#include <capstone/capstone.h>
+
+struct platform {
+	cs_arch arch;
+	cs_mode mode;
+	unsigned char *code;
+	size_t size;
+	const char *comment;
+};
+
+static csh handle;
+
+static void print_string_hex(const char *comment, unsigned char *str, size_t len)
+{
+	unsigned char *c;
+
+	printf("%s", comment);
+	for (c = str; c < str + len; c++) {
+		printf("0x%02x ", *c & 0xff);
+	}
+
+	printf("\n");
+}
+
+static void print_insn_detail(cs_insn *ins)
+{
+	int i;
+	int n;
+	cs_riscv *riscv;
+	cs_detail *detail;
+
+	// detail can be NULL on "data" instruction if SKIPDATA option is turned ON
+	if (ins->detail == NULL)
+		return;
+
+	riscv = &(ins->detail->riscv);
+	detail = ins->detail;
+	if (riscv->op_count)
+		printf("\top_count: %u\n", riscv->op_count);
+
+	for (i = 0; i < riscv->op_count; i++) {
+		cs_riscv_op *op = &(riscv->operands[i]);
+		switch((int)op->type) {
+			default:
+				printf("\terror in opt_type: %u\n", (int)op->type);
+				break;
+			case RISCV_OP_REG:
+				printf("\t\toperands[%u].type: REG = %s\n", i, cs_reg_name(handle, op->reg));
+				break;
+			case RISCV_OP_IMM:
+				printf("\t\toperands[%u].type: IMM = 0x%" PRIx64 "\n", i, op->imm);
+				break;
+			case RISCV_OP_MEM:
+				printf("\t\toperands[%u].type: MEM\n", i);
+				if (op->mem.base != RISCV_REG_INVALID)
+					printf("\t\t\toperands[%u].mem.base: REG = %s\n",
+							i, cs_reg_name(handle, op->mem.base));
+				if (op->mem.disp != 0)
+					printf("\t\t\toperands[%u].mem.disp: 0x%" PRIx64 "\n", i, op->mem.disp);
+
+				break;
+		}
+
+	}
+	
+	//print the groups this instruction belongs to
+	if (detail->groups_count > 0) {
+		printf("\tThis instruction belongs to groups: ");
+		for (n = 0; n < detail->groups_count; n++) {
+			printf("%s ", cs_group_name(handle, detail->groups[n]));
+		}
+		printf("\n");
+	}
+
+	printf("\n");
+}
+
+static void test()
+{
+	//   \xb3\x00\x00\x00 for add ra, zero, zero
+	//	\x93\x00\x21\x00
+	//  \xb3\x70\x31\x00
+#define RISCV_CODE32 "\x37\x34\x00\x00\x97\x82\x00\x00\xef\x00\x80\x00\xef\xf0\x1f\xff\xe7\x00\x45\x00\xe7\x00\xc0\xff\x63\x05\x41\x00\xe3\x9d\x61\xfe\x63\xca\x93\x00\x63\x53\xb5\x00\x63\x65\xd6\x00\x63\x76\xf7\x00\x03\x88\x18\x00\x03\x99\x49\x00\x03\xaa\x6a\x00\x03\xcb\x2b\x01\x03\xdc\x8c\x01\x23\x86\xad\x03\x23\x9a\xce\x03\x23\x8f\xef\x01\x93\x00\xe0\x00\x13\xa1\x01\x01\x13\xb2\x02\x7d\x13\xc3\x03\xdd\x13\xe4\xc4\x12\x13\xf5\x85\x0c\x13\x96\xe6\x01\x13\xd7\x97\x01\x13\xd8\xf8\x40\x33\x89\x49\x01\xb3\x0a\x7b\x41\x33\xac\xac\x01\xb3\x3d\xde\x01\x33\xd2\x62\x40\xb3\x43\x94\x00\x33\xe5\xc5\x00\xb3\x76\xf7\x00\xb3\x54\x39\x01\xb3\x50\x31\x00\x33\x9f\x0f\x00"
+#define RISCV_CODE64 "\x13\x04\xa8\x7a"  // aaa80413
+	struct platform platforms[] = {
+		{
+			CS_ARCH_RISCV,
+			CS_MODE_RISCV32,
+			(unsigned char *)RISCV_CODE32,
+			sizeof(RISCV_CODE32) - 1,
+			"RISCV32G"
+		},
+		{
+			CS_ARCH_RISCV,
+			CS_MODE_RISCV64,
+			(unsigned char *)RISCV_CODE64,
+			sizeof(RISCV_CODE64) - 1,
+			"RISCV64G"
+		}
```

Comment:
is there any sample for mode CS_MODE_RISCV ?

Reply:
RISCV compress mode even RISCV specification is a draft, when LLVM upstream is some stable, I will get time update this capstone riscv port.
so, no, for current.